### PR TITLE
Support timeout for tool

### LINKED ISSUE #45: Add varied timeouts in config.
Add varied timeouts in config.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes #45

### CODE REVIEW

File: tools.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
   },
   "ruby_test_runner": {
     "network": "none",
-    "max_memory": "1GB"
+    "max_memory": "1GB",
+    "timeout": "10"
```
Comment (ErikSchierboom) [inline tools.json:17]: I remembered you saying that the timeout for the Ruby test runner should be lower. Let me know if this is acceptable.

File: lib/tooling_invoker/configuration.rb [hunk 2]
```diff
@@ -37,6 +37,11 @@ def network_for_tool(tool)
       tool_config&.fetch("network", nil) || "none"
     end
 
+    def timeout_for_tool(tool)
+      tool_config = tools_config[tool.tr('-', '_')]
```
Comment (iHiD) [inline lib/tooling_invoker/configuration.rb:41]: You probably wanna DRY this line up now :)

File: tools.json (hunk 1 above)
Comment (iHiD) [reply tools.json:17]: `4` would be fine. 

Any reason not to use a string rather than a number ?

**Comment by iHiD** [comment]
So we also need to use this value.

Best place is probably `lib/tooling_invoker/exec_docker.rb` (that's also where the network config is used).

You can probably remove the `job.timeout_s` method and just use this call instead?

**Pull request merged**